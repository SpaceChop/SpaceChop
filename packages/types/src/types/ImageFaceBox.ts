import * as t from 'runtypes';

const ImageFaceBox = t.Record({
  x: t.Number,
  y: t.Number,
  width: t.Number,
  height: t.Number,
});

export type ImageFaceBox = t.Static<typeof ImageFaceBox>;
export default ImageFaceBox;
