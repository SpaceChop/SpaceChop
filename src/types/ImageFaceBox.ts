import * as t from 'runtypes';

const ImageFaceBox = t.Record({
  height: t.Number,
  width: t.Number,
  x: t.Number,
  y: t.Number,
});

export type ImageFaceBox = t.Static<typeof ImageFaceBox>;
export default ImageFaceBox;
