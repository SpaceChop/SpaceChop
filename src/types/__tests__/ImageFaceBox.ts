import ImageFaceBox from '../ImageFaceBox';

describe('validate ImageFaceBox', () => {
  describe('invalid', () => {
    it('no empty config', () => {
      const config = {};
      expect(() => ImageFaceBox.check(config)).toThrowErrorMatchingSnapshot();
    });
  });
});
