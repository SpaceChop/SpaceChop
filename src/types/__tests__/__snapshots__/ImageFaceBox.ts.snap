// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate ImageFaceBox invalid no empty config 1`] = `"Expected number, but was undefined"`;
